configure.ac

```diff
@@ -4175,6 +4175,32 @@ AS_IF([test "${universal_binary-no}" = yes ], [
 
 [end]_group
 
+AC_MSG_CHECKING([whether ASAN is enabled])
+AC_RUN_IFELSE([AC_LANG_PROGRAM([[]], [[
+#if (defined(__has_feature) && __has_feature(address_sanitizer)) || defined(__SANITIZE_ADDRES__)
+        return 0;
+#else
+        return 1;
+#endif
```

Comment:
Don't use `AC_RUN_IFELSE`.

```suggestion
AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[]], [[
#if (defined(__has_feature) && __has_feature(address_sanitizer)) || defined(__SANITIZE_ADDRES__)
        return 0;
#else
# error no address sanitizer
#endif
```

I don't think this is not needed to do in configure.

Reply:
Ah of course I didn't realise it actually didn't need to run.

I put this in autoconf because it affected multiple files, but we're already having a discussion about whether that's a good idea or not in the other thread. But that's _why_ I put it in autoconf.